src/node/transaction.cpp

```diff
@@ -14,12 +14,12 @@
 
 #include <future>
 
-TransactionError BroadcastTransaction(const CTransactionRef tx, std::string& err_string, const CAmount& max_tx_fee, bool relay, bool wait_callback)
+TransactionError BroadcastTransaction(const CTransactionRef tx, std::string& err_string, const CAmount& max_tx_fee, bool relay, bool wait_callback, CConnman* connman)
```

Comment:
Couldn't BroadcastTransaction be directly a method of Node instead of passing CConnman ? Maybe sendrawtransaction should be given g_rpc_chain instead of g_rpc_node, overall it's seems to use the chain as a server too

Reply:
> Yes, but I don't think that would be a good idea. I think interface classes should be a thin layer on top of wallet and node code and not a place where real functionality is implemented. The model is:

I don't understand this answer. `src/node/transaction.cpp` is "node code" as I understand it and not merely some interface. All the other global state accessed in this function (chainstate, mempool) should eventually be a node member. So the suggestion by @ariard to make this a node member function makes sense to me.

Reply:
Thanks got your point. Was doing far out-of-scope thinking, `sendrawtransaction` to your local node isn't great for privacy, it could be a binary to connect to any node to broadcast instead of a RPC call. But yes, in that case it wouldn't even use any interface..